### PR TITLE
Add Tesla telemetry middleware

### PR DESCRIPTION
**Pull request by DavidTobin**
This PR adds the Tesla telemetry middleware which will allow for metrics tracking against Consulex.

### CODE REVIEW

**Review by Auralcat — COMMENTED** [review]
Looks OK to me, but I think we should wait for @balena's review before merging

File: lib/consul/connection.ex
Comment (balena) [inline lib/consul/connection.ex:45]: Can we make it optional (including the metadata) instead?

**Review by balena — CHANGES_REQUESTED** [review]
Check comment.

**Pull request closed without merging**
